src/main/java/io/jenkins/plugins/traceable/ast/TraceableASTGenerateReportAction.java

```diff
@@ -63,36 +57,13 @@ public void onAttached(Run<?, ?> r) {
 
     private void runScript(String scriptPath, String[] args) {
         try {
-            // Read the bundled script as string
-            String bundledScript = CharStreams.toString(
-                    new InputStreamReader(getClass().getResourceAsStream(scriptPath), Charsets.UTF_8));
-            // Create a temp file with uuid appended to the name just to be safe
-            File tempFile = File.createTempFile("script_" + UUID.randomUUID().toString(), ".sh");
-            // Write the string to temp file
-            BufferedWriter x = Files.newWriter(tempFile, Charsets.UTF_8);
-            x.write(bundledScript);
-            x.close();
-            String execScript = new StringBuffer()
-                    .append("/bin/bash ")
-                    .append(tempFile.getAbsolutePath())
-                    .toString();
-            for (int i = 0; i < args.length; i++) {
-                if (args[i] != null && !args[i].equals("")) execScript += " " + args[i];
-                else execScript += " ''";
-            }
-            ProcessBuilder processBuilder = new ProcessBuilder(execScript);
-            processBuilder.redirectErrorStream(true);
-            Process pb = processBuilder.start();
-            logOutput(pb.getInputStream());
-            pb.waitFor();
-            int reportCmdExitValue = pb.exitValue();
-            boolean deleted_temp = tempFile.delete();
-            if (reportCmdExitValue == 4) {
+            String[] status = this.workspace.act(new GenerateReport(scriptPath, args, run.getId()));
+
+            if (status[0].equals("FAILURE")) {
```

Comment:
https://github.com/Traceableai/traceable-xast-jenkins-plugin/blob/ac528bd84496023d458123372cdf5715280af158/src/main/java/io/jenkins/plugins/traceable/ast/scan/utils/GenerateReport.java#L46

Same as before, we are explicitly declaring an array of size 2. Also the value of `processExitValue` is `SUCCESS` by default. So it will never be null. It will either be `SUCCESS` or `FAILURE`